torch/_inductor/lowering.py

```diff
@@ -3556,20 +3577,68 @@ def fn(idx, return_index):
         else:
             return maxval
 
-    r1 = Pointwise.create(
+    out = Pointwise.create(
         device=x.get_device(),
         dtype=x.get_dtype(),
         inner_fn=functools.partial(fn, return_index=False),
         ranges=new_size,
     )
-    r2 = Pointwise.create(
+    offsets = Pointwise.create(
         device=x.get_device(),
-        dtype=torch.int64,
+        dtype=torch.int8,
         inner_fn=functools.partial(fn, return_index=True),
         ranges=new_size,
     )
-    # TODO(jansel): should we force these to be realized?
-    return r1, r2
+    return out, offsets
+
+
+@register_lowering(aten.max_pool2d_with_indices, type_promotion_kind=None)
+def max_pool2d_with_indices(
+    x, kernel_size, stride=None, padding=0, dilation=1, ceil_mode=False
+):
+    # assert we are on the fallback path, otherwise we should use the the
+    # prims._low_memory variant
+    kernel_size, stride, padding, dilation, _ = max_pool2d_checks(
+        x, kernel_size, stride, padding, dilation, assert_fallback=True
+    )
+    return fallback_max_pool2d_with_indices(
+        x, kernel_size, stride, padding, dilation, ceil_mode
+    )
```

Comment:
Can you not use `make_fallback` here?

Reply:
Yes, I wanted the assert here temporarily to make sure I had covered paths like `adaptive_max_pool2d`.